Twenty Thirteen: fix "Older Posts" label for archive pages with respect to order

shrivastavanolo opened the pull request:
<!--
Hi there! Thanks for contributing to WordPress!

Pull Requests in this GitHub repository **must** be linked to a ticket in the WordPress Core Trac instance (https://core.trac.wordpress.org), and are only used for code review. **No pull requests will be merged on GitHub.**

See the WordPress Handbook page on using PRs for Code Review more information: https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/

If this is your first time contributing, you may also find reviewing these guides first to be helpful:
- FAQs for New Contributors: https://make.wordpress.org/core/handbook/tutorials/faq-for-new-contributors/
- Contributing with Code Guide: https://make.wordpress.org/core/handbook/contribute/
- WordPress Coding Standards: https://make.wordpress.org/core/handbook/best-practices/coding-standards/
- Inline Documentation Standards: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/
- Browser Support Policies: https://make.wordpress.org/core/handbook/best-practices/browser-support/
- Proper spelling and grammar related best practices: https://make.wordpress.org/core/handbook/best-practices/spelling/
-->

<!-- Insert a description of your changes here -->
Similar to [PR](https://github.com/WordPress/wordpress-develop/pull/9079), [PR](https://github.com/WordPress/wordpress-develop/pull/9082), [PR](https://github.com/WordPress/wordpress-develop/pull/9089)
Fixes the issue: "Older Entries" and "Newer Entries" links are wrong when entries displayed in ascending order for the **Twenty Thirteen** theme by adding dynamic labels according to the order query var (DESC or ASC). Also updated the div classes (These classes controlled the position of the labels) for the nav labels to switch according to the order. For context:

### Without switching classes
<img width="1470" alt="Screenshot 2025-06-27 at 1 17 26 PM" src="https://github.com/user-attachments/assets/19e09591-244f-4a4f-8f03-8a22d2623bda" />

### Switching classes
<img width="1468" alt="Screenshot 2025-06-27 at 1 16 36 PM" src="https://github.com/user-attachments/assets/bfa48a45-ff17-4389-b174-5b0905678c30" />

---

Trac ticket: https://core.trac.wordpress.org/ticket/10219

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

sabernhardt commented:
For Twenty Thirteen and older themes, `nav-previous` still needs to be in the markup before `nav-next` so the visual order and DOM order match. (See [Meaningful Sequence](https://www.w3.org/TR/WCAG22/#meaningful-sequence) and [Focus Order](https://www.w3.org/TR/WCAG22/#focus-order).)

I also do not think Twenty Thirteen should add the 10-pixel margin on the opposite side of the `meta-nav` arrows.

shrivastavanolo committed:
fix: switch nav labels while preserving DOM order

shrivastavanolo committed:
refactor: remove redundant checks and add escaping

shrivastavanolo commented:
Refactored to remove redundant checks using `$is_desc` and preserved the .nav-previous / .nav-next DOM order. Please let me know if this looks good, I’ll follow the same pattern in the PRs for the older themes as well. Thanks!

cc: @sabernhardt

shrivastavanolo committed:
refactor: replace wp_kses with wp_kses_post

sabernhardt reviewed (approved):
This makes sense to me for Twenty Thirteen.

With `?order=ASC`, the links' `href` attributes switch, without changing the appearance. The following `diff` shows the link changes on the `ASC` page, before and after applying the patch, within the markup:

```diff
-					<nav class="navigation paging-navigation">
+			
+		<nav class="navigation paging-navigation">
 		<h1 class="screen-reader-text">
 			Posts navigation		</h1>
 		<div class="nav-links">
-
-						<div class="nav-previous"><a href="http://localhost/svn/src/tag/alice/page/3/?order=ASC" ><span class="meta-nav">&larr;</span> Older posts</a></div>
-			
-						<div class="nav-next"><a href="http://localhost/svn/src/tag/alice/?order=ASC" >Newer posts <span class="meta-nav">&rarr;</span></a></div>
-			
+					<div class="nav-previous">
+				<a href="http://localhost/svn/src/tag/alice/?order=ASC"><span class="meta-nav">&larr;</span> Older posts</a>			</div>
+						<div class="nav-next">
+				<a href="http://localhost/svn/src/tag/alice/page/3/?order=ASC">Newer posts <span class="meta-nav">&rarr;</span></a>			</div>
+		
 		</div><!-- .nav-links -->
 	</nav><!-- .navigation -->
```

In default descending order, the markup has different spacing between elements, and `wp_kses_post()` removes the space after the `href` values. Otherwise, it is the same as before the patch.

```diff
-					<nav class="navigation paging-navigation">
+			
+		<nav class="navigation paging-navigation">
 		<h1 class="screen-reader-text">
 			Posts navigation		</h1>
 		<div class="nav-links">
-
-						<div class="nav-previous"><a href="http://localhost/svn/src/tag/alice/page/3/" ><span class="meta-nav">&larr;</span> Older posts</a></div>
-			
-						<div class="nav-next"><a href="http://localhost/svn/src/tag/alice/" >Newer posts <span class="meta-nav">&rarr;</span></a></div>
-			
+					<div class="nav-previous">
+				<a href="http://localhost/svn/src/tag/alice/page/3/"><span class="meta-nav">&larr;</span> Older posts</a>			</div>
+						<div class="nav-next">
+				<a href="http://localhost/svn/src/tag/alice/">Newer posts <span class="meta-nav">&rarr;</span></a>			</div>
+		
 		</div><!-- .nav-links -->
 	</nav><!-- .navigation -->
```

The navigation appears the same as it did before the patch, regardless of the posts' order. The screenshots show the theme's `:focus` outline for the first link in page navigation (DOM order).

<img width="700" height="140" alt="with patch, ascending order, English" src="https://github.com/user-attachments/assets/7b3a51cd-8e87-47c7-a97e-5cf7a1002164" />

I also checked French and Hebrew.

<img width="700" height="140" alt="with patch, ascending order, French" src="https://github.com/user-attachments/assets/a79cabd7-def1-4be8-92e0-84791a2769c2" />
<img width="700" height="140" alt="with patch, ascending order, Hebrew" src="https://github.com/user-attachments/assets/92c82502-6d39-481f-9442-852a61e0936b" />

----

Full-page screenshots show three posts, with different published dates (matching the chapter number to the day of the month).
- [Before patch, default order](https://github.com/user-attachments/assets/3c7a042f-8856-4381-b3a3-c534c1f1fb19)
- [Before patch, ascending order](https://github.com/user-attachments/assets/4f6a2b94-5255-45fe-bd65-9a1484426e6e)
- [With patch, default order](https://github.com/user-attachments/assets/913d8cf8-1f90-4585-bdeb-1513d86a9d11)
- [With patch, ascending order](https://github.com/user-attachments/assets/62570b5a-14a7-4d36-9d3f-1cf7ec96a002)

Twenty Thirteen does not have any adjustment for links on small screens.
<img width="400" height="300" alt="navigation on small screens, with patch, in English" src="https://github.com/user-attachments/assets/97d29545-5802-48cc-b93f-4c5c8f5529b5" />

I also ran each translation through `wp_kses_post()` to make sure they all display properly.
- [Left-to-right languages](https://github.com/user-attachments/assets/f92c6b9d-f53c-43eb-b2a3-75f17225d952)
- [Right-to-left languages](https://github.com/user-attachments/assets/d2e8b238-6167-4ece-90e6-ad096e2c00fa)

shail-mehta commented:
|Before|After|
|-|-|
|<img width="400" height="400" alt="twenty-thirteen-before" src="https://github.com/user-attachments/assets/8e7c15f6-7269-4f44-849c-b305e6450562" /> | <img width="400" height="400" alt="twenty-thirteen-after" src="https://github.com/user-attachments/assets/a5f2855c-488e-43b1-af5d-bad272db498b" /> |

joedolson committed:
Merge branch 'trunk' into pr/9090

joedolson committed:
Remove `wp_kses_post()`

This isn't necessary, since the content is already escaped in the generating function.

Late escaping is a good practice, but not sure this is necessary.
